### PR TITLE
Safer lookup and generate static resources for web dep locator

### LINKED ISSUE #44126: Resources should be found from the classpath instead in web dep locator
Currently, resources are found from the output directory which can be really fragile:
https://github.com/quarkusio/quarkus/blob/main/extensions/web-dependency-locator/deployment/src/main/java/io/quarkus/webdependency/locator/deployment/WebDependencyLocatorProcessor.java#L51

Instead we should use:
 `ClassPathUtils.consumeAsPaths(config.webRoot)`

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #44126

### CODE REVIEW

**Comment by ia3andy** [comment]
This PR also fixe a wrong directory in the doc cc @tqvarnst

**Comment by tqvarnst** [comment]
This issue is kind of annoying, and to work around it, I have to CTRL-C out of dev mode and start dev mode again, so it would be nice to have this backported to 3.15 as well.

**Comment by aloubyansky** [comment]
This is a breaking change, isn't it? The new code doesn't look in the project resources at all?
BTW, the previous code made some assumptions that it shouldn't do.

File: bom/application/pom.xml [hunk 1]
```diff
@@ -53,7 +53,7 @@
         <smallrye-metrics.version>4.0.0</smallrye-metrics.version>
         <smallrye-open-api.version>3.13.0</smallrye-open-api.version>
         <smallrye-graphql.version>2.11.0</smallrye-graphql.version>
-        <smallrye-fault-tolerance.version>6.5.0</smallrye-fault-tolerance.version>
```
Comment (aloubyansky) [inline bom/application/pom.xml:56]: Given this is a candidate for backporting to 3.15, here is a dependency version upgrade

Comment (aloubyansky) [reply bom/application/pom.xml:56]: Oh, I accidentlly pushed this.

**Comment by aloubyansky** [comment]
@ia3andy I pushed an enhancement to make it easier to process runtime resources as the first commit

**Comment by ia3andy** [comment]
thanks @aloubyansky this is pretty cool! I will use the same for Roq :)

**Comment by ia3andy** [comment]
@aloubyansky could you approve this

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**